### PR TITLE
Added epsilons to overdispersion in gCNV models to avoid NaNs.

### LINKED ISSUE #4824: gCNV nan errors
@samuelklee @asmirnov239 @mbabadi I tried to run a 30-sample cohort through gCNV on all canonical chromosomes with 250bp bins sharded in 10k-interval blocks, but PostprocessGermlineCNVCalls gave the following error:

```...
19:26:14.967 INFO  PostprocessGermlineCNVCalls - Analyzing shard 223...
19:26:15.107 INFO  PostprocessGermlineCNVCalls - Analyzing shard 224...
19:26:15.259 INFO  PostprocessGermlineCNVCalls - Analyzing shard 225...
19:26:15.260 INFO  PostprocessGermlineCNVCalls - Shutting down engine
[May 29, 2018 7:26:15 PM UTC] org.broadinstitute.hellbender.tools.copynumber.PostprocessGermlineCNVCalls done. Elapsed time: 3.34 minutes.
Runtime.totalMemory()=39753089024
***********************************************************************

A USER ERROR has occurred: Bad input: Validation error occurred on line %d of the posterior file: Posterior probabilities for at at least one posterior record do not sum up to one.
```

After inspecting the output from shard 225, it seems that the model starts producing nan values after ~1600 warmup iterations (looking at the ELBO log). This shard corresponds to a pericentromeric region chr3:91540501-94090250. 

It would be nice to have the option to bypass this error in PostprocessGermlineCNVCalls.

Here is the model config for the shard:

```"p_alt": 1e-06,
 "p_active": 0.01,
 "cnv_coherence_length": 10000.0,
 "class_coherence_length": 10000.0,
 "max_copy_number": 5,
 "num_calling_processes": 1,
 "num_copy_number_states": 6,
 "num_copy_number_classes": 2
 "max_bias_factors": 5,
 "mapping_error_rate": 0.01,
 "psi_t_scale": 0.001,
 "psi_s_scale": 0.0001,
 "depth_correction_tau": 10000.0,
 "log_mean_bias_std": 0.1,
 "init_ard_rel_unexplained_variance": 0.1,
 "num_gc_bins": 20,
 "gc_curve_sd": 1.0,
 "q_c_expectation_mode": "hybrid",
 "active_class_padding_hybrid_mode": 50000,
 "enable_bias_factors": false,
 "enable_explicit_gc_bias_modeling": false,
 "disable_bias_factors_in_active_class": false
 "version": "0.7"
```

### LINKED ISSUE #6226: Get rid of python warning in gCNV code.
Just a minor annoyance:

```/home/slee/.pyenv/versions/anaconda3-5.3.1/envs/gatk/lib/python3.6/site-packages/gcnvkernel/io/io_commons.py:394: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  mu_slice = mu_all[_get_singleton_slice_along_axis(mu_all, var_sample_axis, sample_index)]
/home/slee/.pyenv/versions/anaconda3-5.3.1/envs/gatk/lib/python3.6/site-packages/gcnvkernel/io/io_commons.py:395: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  std_slice = std_all[_get_singleton_slice_along_axis(mu_all, var_sample_axis, sample_index)]```

### LINKED ISSUE #6227: Fix typo in gcnvkernel setup.py file.
Theano version should be 1.0.4.  Doesn't have any effect unless trying to install the package manually with pip.

### PR DESCRIPTION
**Pull request by samuelklee**
Closes #4824 (at least for underflow of overdispersion)
Closes #6226 
Closes #6227

Along with some other minor fixes.

I've done some manual testing and this change is most likely harmless, but ideally we'd have some better automated testing to cover this sort of minor model change...  Extremely conscientious users might want to rebuild models just to be safe, which we can mention in the release notes.

### CODE REVIEW

**Comment by samuelklee** [comment]
@mwalker174 @asmirnov239 I probably should rebuild models used for test resources, etc.  Hopefully can get this done by release on Thursday.  Might also fix #5852 while I'm at it.

**Comment by samuelklee** [comment]
Turns out that the models didn't change for the WDL tests.  @mwalker174 @asmirnov239 could one of you review?

**Comment by samuelklee** [comment]
@droazen Ideally this and #6244 would be in before release, thanks!

**Comment by samuelklee** [comment]
Thanks @mwalker174!  That setup_gcnvkernel.py file is just for manual installations (users or devs can rename it to setup.py and install the package using pip, for example---I think there's some documentation on this in the readme in the python folder.).  So it's not checked by any build tests.

File: src/main/python/org/broadinstitute/hellbender/gcnvkernel/models/model_denoising_calling.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
 
 _logger = logging.getLogger(__name__)
 
+_eps = 1E-10
```
Comment (asmirnov239) [inline src/main/python/org/broadinstitute/hellbender/gcnvkernel/models/model_denoising_calling.py]: Could you extract this constant here and in `model_ploidy.py` to a separate file? Maybe it's also would be worth promoting this to a config parameter so we can experiment with it in the future

Comment (samuelklee) [reply src/main/python/org/broadinstitute/hellbender/gcnvkernel/models/model_denoising_calling.py]: Just put it in models.commons.  Maybe we can think about exposing it if more NaNs crop up.

**Pull request merged**